### ISSUE TITLE
App crashes durring loading thumbnails on mobile app

### ISSUE THREAD

**Issue by al73rus** [immich-app/immich#11113]
### The bug

When you open immich app on android and the thumbs load up, and you start scrolling down in my case 5-7days of photos (had vacation) the app first freezes and then crash itself. It oooks like not enough ram or other thing. Have nothing in logs in the app. It just goes down and you have to scroll again. When doing it slowly i can go maybe a month back before it crashes.

### The OS that Immich Server is running on

Unraid

### Version of Immich Server

v1.108.0

### Version of Immich Mobile App

v1.108.0 build 148

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
None
```

### Your .env content

```Shell
None
```

### Reproduction steps

```bash
1.open app on adroid
2.scroll down so those thumbs get loaded
3. After sometime app starting to freeze and eventualy crash
...
```

**Comment by alextran1502** [comment]
Hello, what is the model of your phone?

**Comment by al73rus** [comment]
Poco x3 nfc. I didn't had those problem in earlier apps.

**Comment by alextran1502** [comment]
And how many assets do you have on your instance?

**Comment by al73rus** [comment]
[Immich_log_2024-07-15T153849.440283 (1).log](https://github.com/user-attachments/files/16236391/Immich_log_2024-07-15T153849.440283.1.log)

**Comment by al73rus** [comment]
It stsrted from the latest version. I didn't notice it before. I return from my vacation and was showing photos from my and my wifes phone which where uploaded to my server.
And when i do it one by one it works normaly. Only when i scroll down and the phone can't kerp up with the loadinh then it freezes and crash.
I have around 25k assets. But usualy i don't scroll further than a week back unless i need to find a photo from way back.
I changed the storage template so i could redownload photos to my pc.
Also i regenerated all thumbs bc some where broken after the move.
Also i added around 2-3k assets few days back (older photos way back in 2012-2016.
The eeb page also have some lag dhowing those thumbs now when i scroll it. Some are loaded and some need a long time.

**Comment by al73rus** [comment]
Just checked the web interface and it also lags. It doesn't matter if it's on local or remote. Some thumb load and some are blured and don't load after time. When i click the photo it loads normaly.

If i need to run some comands to verify anything or record a vid how it happends and my setting just let me know

**Comment by Craeckie** [comment]
I have the same problem on the same phone since a recent update.
For me, the OS stutters and either other apps or Immich crashes first.

The memory usage in the Developer Tools shows a low average, but 3.6Gb of maximum usage, so I also suspect it's because of RAM usage of Immich.

I use Lineage OS 21, I have 310Gb, ~33k media files

**Comment by al73rus** [comment]
One thing i noticed is thst the app crashes mostly on uploaded assets. Those that i have on my phone works hood but not perfect, what i mean is the app dorsn't crash, also when i turn off shoeing assets from my wife in my galery it also somehow works but found another wierd behavior.. when you "zoom/select" asset you wanna enlarge and go back the thumbs are regenerated from start.
I'll record a vid and upload it on yt later. Busy with kids atm.

**Comment by alextran1502** [comment]
It could be from this PR https://github.com/immich-app/immich/pull/11002. Let me try to build an APK without that config and you guys can try loading it on your device

**Comment by alextran1502** [comment]
Can you guys try this APK? https://drive.google.com/file/d/1-RR36jntGrqSf25X26G7ZFNYZ0_YPvnF/view?usp=sharing

**Comment by alextran1502** [comment]
> Just checked the web interface and it also lags. It doesn't matter if it's on local or remote. Some thumb load and some are blured and don't load after time. When i click the photo it loads normally.

This indicates something is wrong with your mount point

**Comment by al73rus** [comment]
Maybe thebstorage template or moving those photo made a mess with that.
Alao about the apk. Instaled and it crashes my system innfew seconds just after loading up those assets.
One thing i remeber is changing my org admin name to maro in the storate template on users screen and moving all asseta after that.

**Comment by al73rus** [comment]
Even if i try to log in localu the app freezes even before i can open an asset.
Is there a comand or can i manualy delete thumbs to regenerate them a new like i would just upload it from the start?

**Comment by al73rus** [comment]
I resteted my docker with database on unraid. And nowni'm generating new thumbnails. 
So for the next 20-30 mins i won't be able to tell if it changes anything.

**Comment by al73rus** [comment]
Still no change after what i did. Gonna reinstall the app feom store tomorrow. And see if it heals itself or will have to wait for some new updates. Will keep this updated.
Web pages is running smoothly now but got a 1k thumbs error which i need to fix. Maybe i'll re install from scratch and reupload the photoa on the server and force clean my phone with factory reset.

**Comment by Craeckie** [comment]
I tried with your 

> Can you guys try this APK? https://drive.google.com/file/d/1-RR36jntGrqSf25X26G7ZFNYZ0_YPvnF/view?usp=sharing

I tried with this apk and the problem is still there. I scroll down and then wait for the thumbnails. Usually, either Immich or Tailscale crashes. If TS crashes, my phone looses connection and the logs of Immich are full of network errors. I couldn't find any other kind of error in the logs.

**Comment by al73rus** [comment]
I found a new thing maybe related to the crashes.
The app after loading around 40-60 thumbs when scrolling took around 2.5-3 GB of ram and if i scroll a bit more it goes up to 4 and it crashes if (the whole system reboots and loada back up bc of lack of ram)

**Comment by alextran1502** [comment]
@Craeckie what is your phone model and how many asset do you have on your instance?

**Comment by al73rus** [comment]
After phone reset nothing changed (either way i had to).
Found another bug or error:
When you open immich and it loads up thumbs with the timeline it loads up all thumbs into ram even if you don't watch it. Also when you open a photo to enlarge it and go back by pressing the < or the back button on the phone it loads all the thumbs from scratch which takes even more ram from my phone. Another thing i noticed that whwn you go out by presing square to see running apps and go back it restarts the whole app.
Maybe in this direction you could find what's wrong.

**Comment by alextran1502** [comment]
@al73rus, Can you use the APK in the 1.107 release and see if it fixes the issue? If it does, then I will check the changes between the two versions to see what could cause this. 

Link to the release here https://github.com/immich-app/immich/releases/tag/v1.107.2

**Comment by al73rus** [comment]
> @al73rus, Can you use the APK in the 1.107 release and see if it fixes the issue? If it does, then I will check the changes between the two versions to see what could cause this.
> 
> Link to the release here https://github.com/immich-app/immich/releases/tag/v1.107.2

I did try the version 107.2 and even 107.1 with no change what so ever it even crashes even more. Eats up a lot more RAM at start up.
Went back to 108 and disabled the timeline to chceck if it changes anything. Sometimes it does help a bit. So i can scroll some more down the line.
I'm uploading a vid to show you when the app starts and how much ram it takes after i scroll a bit.

https://youtu.be/1m_lHpKhcQg?si=hBBIwTr83jEpitpb

Also it shows how the app behaves as U go out of it and back again when it loads up assets to show again.

Last thing i did on the server is to add around 4-6k photos way back before 2016 which i found i my back up folder on the server.

**Comment by al73rus** [comment]
@alextran1502 we can close this issue. Found the problem.
it took some time butni found the real problem.

When i deleted all the photos and accounts and set them up and reuploaded assets from scratch (with proper storage template from the start) it now works as intended no ram eating no crashes all good.
I think the wholestorage migration and thumbs migration should be double checked.
It all startet with that as i needed to get the assets have a copy of them organized.

**Comment by Craeckie** [comment]
I also tried with 106.4 and 108.0, both from F-Droid, and have the same problem.
I didn't do any storage migration, but I had trouble with thumbnails not being found before this issue.
I tried to re-create all thumbnails and ran some repair function, but apparently this didn't fix it.

**Comment by Saschl** [comment]
Seeing a similar issue here: https://github.com/immich-app/immich/issues/11258 (although on iOS, let me know if the issues should be merged). 
I also applied storage templates and see stutters in the WebUI (mainly Edge/Chrome, Safari seems fine also on my iPad)

**Comment by drunkleee** [comment]
I have the same problem,the main problem happen in my iphone, when the thumbnail no load up totally.

**Comment by alextran1502** [comment]
@drunkleee what is your iPhone model? Can you explain a bit more?

**Comment by drunkleee** [comment]
When I just poured the photos into immich and Immich was still generating thumbnails, my iPhone always crashed when using immich. However, my thumbnail task has been completed now, and there seems to be no crash problem.

> @drunkleee what is your iPhone model? Can you explain a bit more?

**Comment by Cookie-Monster-Coder** [comment]
I have the same problem with a Samsung 22+. Have 14k assets.

**Comment by mauropriori** [comment]
Same problem with Samsung a52s even with only 100 assets

**Comment by Matou25** [comment]
Same for me, on s21+ or Huawei P30pro.
App freeze after variable time when loading or scrolling or viewing photos.
I have 62k pictures.
4G or wifi, sale freeze.
Nothing in the app logs

**Comment by alextran1502** [comment]
@Matou25 do you have 62k photos on the server or on your local device?

**Comment by Matou25** [comment]
@alextran1502 on the server

**Comment by alextran1502** [comment]
@Matou25 Are you still seeing the spinner icon at the bottom app bar when you are on the Photos page?

**Comment by Matou25** [comment]
It dépends, some exemple :
- I open the app clic on the 1st picture, and view photos by clic on right arrow, après 30-45s app freeze.
- I open the app, clic search then choose the 1st person, somtime I see alla the pictures, star viewing and it crash
- I open the app, clic search then choose the 1st person, circle turn then stop... App is freeze

**Comment by Matou25** [comment]
@alextran1502
No spining icon when open app.
I can see this icon after: open app, clic on search, return on photo... It spin then freeze

**Comment by alextran1502** [comment]
@Matou25 Can you go to the app settings > Advanced > enable the prefer remote image button?

**Comment by Matou25** [comment]
One more info, all my asset are in an external library

**Comment by Matou25** [comment]
> @Matou25 Can you go to the app settings > Advanced > enable the prefer remote image button?

@alextran1502 done, same freeze

**Comment by ok1ok1** [comment]
I have a lot of assets. (I think its a lot) Over a half million all in external libraries.  I've experienced many of these same problems.

I went weeks trying to use the app as soon as the first set of thumbnails loaded and it never quite worked out. Initially everything would work okay but the photo & library tab spinners were constantly going. Ultimately the app would slow down, thumbs wouldn't load past blurred versions if at all, then a crash. This progression might take minutes or over an hour. I also experienced issues with search functionality and viewing shared libraries. For me these symptoms were universal across all my mobile devices. 

 Having some patience ended up fixing these issues for me. Try the following. 

1. Make sure that your server isn't under load with import tasks.
2. Enable prefer remote image in advanced settings. (helped some of my older devices)
3. Open the app and don't do any browsing until the spinner on the photos tab goes away. This can take several minutes but it's key. (at least is was for me)

After waiting for the spinner to stop I haven't had a problem since. Subsequent launches of the app load more quickly, the spinners don't run nearly as long. Even my lowest spec devices work quite well.

Give it a try.

**Comment by alextran1502** [comment]
@ok1ok1 Thanks for the reporting. The initial data sync is somewhat intensive if you have loads of assets and it competes with the UI thread, lead to janking, so there is more work on that end to optimize it

**Comment by Cookie-Monster-Coder** [comment]
@ok1ok1 That does not seem to work for me. I wait for the spinner to complete and start scrolling it works fine for a little bit then it starts to get laggy and the thumbnails take very long time to load then crashes the whole phone UI.

**Comment by alextran1502** [comment]
@Cookie-Monster-Coder is that even with Prefer Remote Image enabled?

**Comment by Cookie-Monster-Coder** [comment]
> @Cookie-Monster-Coder is that even with Prefer Remote Image enabled?

Yup, I tried every possible combination I could think of so far and it still crashes. I even clear the cache between each test.

**Comment by ok1ok1** [comment]
> @ok1ok1 That does not seem to work for me. I wait for the spinner to complete and start scrolling it works fine for a little bit then it starts to get laggy and the thumbnails take very long time to load then crashes the whole phone UI.

Did you happen to wait for the section at the top of the photos tab to load? The one with  larger thumbnails of automatically generated collections. They typically are labeled 1 year ago, 2 years ago, etc...

**Comment by alextran1502** [comment]
@Cookie-Monster-Coder How many assets do you have on your instance?

**Comment by Cookie-Monster-Coder** [comment]
Yes, they load instantly for me. @ok1ok1 

Samsung 22+. Have 14k assets. @alextran1502

**Comment by Cookie-Monster-Coder** [comment]
It also happens when searching for something and scrolling through the results.

**Comment by andresilveiraes** [comment]
Hi, I'm testing Immich on Ubuntu desktop and facing the same issue as reported by others on this thread.

I installed it through Casaos and imported around 900gb of Photos (58k photos) 

I can open the Android app but as when scrolling down, it crashes and closes without any error msg.

I observed that the app is loading all the thumbnails 

I use a Samsung Galaxy S21 FE.

**Comment by r-goto** [comment]
The same issue is happening on my iPhone and iPad Pro. (iPhone 12 & iPad Pro 2019model)
Just load immich with a bunch of photos (above 100GB of pics) and open up the app and scroll down a bit for few seconds, it crashes. 

It is not intermittent. It is reproducible 100% every time I open up the app, meaning I cannot scroll and see my photos on my phone and table.

**Comment by r-goto** [comment]
Okay so it happens only with "Admin" account. 
If I create a non-admin account and load up the same photos/videos, the crash doesn't occur. 

But if I login with "admin" account, it crashes on my iPhone and iPad Pro. (iPhone 12 & iPad Pro 2019model)

Hope this info helps to identify what's causing the crash and where the bug is....

**Comment by Matou25** [comment]
@alextran1502  this issues disapear few release ago ( sorry can't say when exactly) , but it come back with V1.130.3 
i try to revert to v1.129 on android - > same freeze after arround 30 sec
i revert my server to v1.129.0 and it work great , even with v1.130 android app

**Comment by alextran1502** [comment]
@Matou25 I don't think we have done anything differently to fix or cause a regression. The only difference would be new assets on your device that might have caused the crash. If you go to App Settings > Advanced > Prefer Remote Image, see if it makes a difference.

**Comment by Matou25** [comment]
No new asset (I dont sync from m'y Phone), i'm only in read mode. 
I can reproduce.
IfI restart my stack with v1.130.3 it freeze again

**Comment by alextran1502** [comment]
Just to confirm that, server version 1.129.0 works, and server version 1.130.3 doesn't work?

**Comment by alextran1502** [comment]
Any thoughts, @shenlong-tanwen?

**Comment by shenlong-tanwen** [comment]
> Any thoughts, [@shenlong-tanwen](https://github.com/shenlong-tanwen?rgh-link-date=2025-03-28T18%3A20%3A53.000Z)?

There are certainly things that should be optimised in the mobile app to improve the performance of the timeline. But I don't think there is any change made in the server that could fix the issue or introduce it again. Android crash dumps will be useful to analyse such issues as these might not be captured in our logger. An app like [LogFox](https://github.com/F0x1d/LogFox) can capture these but it requires a bit of prerequisite to capture them

**Comment by Matou25** [comment]
> Just to confirm that, server version 1.129.0 works, and server version 1.130.3 doesn't work?

Yes

**Comment by Matou25** [comment]
> > Any thoughts, [@shenlong-tanwen](https://github.com/shenlong-tanwen?rgh-link-date=2025-03-28T18%3A20%3A53.000Z)?
> 
> There are certainly things that should be optimised in the mobile app to improve the performance of the timeline. But I don't think there is any change made in the server that could fix the issue or introduce it again. Android crash dumps will be useful to analyse such issues as these might not be captured in our logger. An app like [LogFox](https://github.com/F0x1d/LogFox) can capture these but it requires a bit of prerequisite to capture them

I Will try to give you some log
I Will try holder versionn off the app to check if it work, to confirme or not it's server change

**Comment by Craeckie** [comment]
I just downgraded to 1.129.0, and it is reproducible for me. The recording stops when Immich crashes, while also causing apps in the background to close.
Now that I switched to a Pixel 7 Pro, I need to scroll longer until it crashes, possibly because it has more RAM.
https://github.com/user-attachments/assets/f5a1c7ab-4dbd-4231-abab-cbd5a576dc2c

**Comment by Craeckie** [comment]
Here's the full log.
The lines that are repeated the most often:

```
160 D ExifInterface: No image meets the size requirements of a thumbnail image.                                                 
252 D HWUI    : --- Failed to create image decoder with message 'unimplemented'                                                
1618 W HWUI    : Image decoding logging dropped!                4570 I flutter : Warning database has been locked for 0:00:10.000000. Make sure you always use the transaction object for database operations during a transaction
```
The number at the beginning shows, how often this line was repeated. The log ends when the app crashes (or is closed by Android).

How long I need to scroll until the app crashes seems to depend on the grid size. With smaller thumbnails, it's quicker.

[Immich log 98398357942f.txt](https://github.com/user-attachments/files/19517258/Immich.log.98398357942f.txt)
